### PR TITLE
Create plain-text transform to cover the HipChat and Flowdock transform

### LINKED ISSUE #124: Refactor transform to cover all adapters that use plain-text responses
In order to avoid lots of duplicate code, we need to create a transform to cover HipChat and Flowdock adapters.

### PR DESCRIPTION
**Pull request by llvieira**
Closes #124.

- Created a plain-text-transform to cover the HipChat and Flowdock transform.
- Fixed some bugs: res.count changed to res.members.length and fixed send() method in the plain-text.

### CODE REVIEW

File: src/listener/alerts-listener.js [hunk 1]
```diff
@@ -54,7 +54,7 @@ export default class AlertsListener extends Listener {
     let time = msg.time;
 
     this.client.Alerts.getFilteredAlerts(status, state, time).then((res) => {
-      if (res.count === 0) {
+      if (res.members.length === 0) {
```
Comment (danielreed) [inline src/listener/alerts-listener.js]: Why change this here?  Using the res.count is probably fine.

File: src/listener/ov-listener.js [hunk 2]
```diff
@@ -62,7 +62,15 @@ export default function(robot, client) {
 
     let checkedMessage = filter.check(resource);
     if (typeof checkedMessage !== 'undefined' && checkedMessage.length > 0) {
-      transform.messageRoom(client.notificationsRoom, resource.resource);
+      let room = '';
```
Comment (danielreed) [inline src/listener/ov-listener.js]: Can you shorten this to:

let room = '#' + client.notificationsRoom;
if (robot.adapterName === 'flowdock') {
  room = client.notificationsRoom;
}

File: src/listener/server-hardware.js [hunk 3]
```diff
@@ -179,14 +179,15 @@ export default class ServerHardwareListener extends Listener {
     let status = msg.status.toLowerCase();
     status = status.charAt(0).toUpperCase() + status.slice(1);
     this.client.ServerHardware.getHardwareByStatus(status).then((res) => {
-      if (res.count === 0) {
-        return this.transform.text(msg, msg.message.user.name + ", I didn't find any blades with a " + msg.status.toLowerCase() + " status.");
+
+      if (res.members.length === 0) {
```
Comment (danielreed) [inline src/listener/server-hardware.js]: Do we need to change line 183?

File: src/listener/server-hardware.js [hunk 4]
```diff
@@ -179,14 +179,15 @@ export default class ServerHardwareListener extends Listener {
...
+        return this.transform.text(msg, msg.message.user.name + ", I didn't find any blades with a " + status.toLowerCase() + " status.");
       }
       else {
-        if (msg.status.toLowerCase() === "ok") {
-          return this.transform.send(msg, res, "Okay " + msg.message.user.name + ", the following blades have an " + msg.status.toUpperCase() + " status.");
+        if (status.toLowerCase() === "ok") {
+          return this.transform.send(msg, res, "Okay " + msg.message.user.name + ", the following blades have an " + status.toLowerCase() + " status.");
```
Comment (danielreed) [inline src/listener/server-hardware.js]: Why lower case on the status in the message to the user?

File: src/listener/server-hardware.js [hunk 5]
```diff
@@ -198,7 +199,7 @@ export default class ServerHardwareListener extends Listener {
     let status = msg.powerState.substring(8, msg.powerState.length);
     status = status.charAt(0).toUpperCase() + status.slice(1);
     this.client.ServerHardware.getHardwareByPowerState(status).then((res) => {
-      if (res.count === 0) {
+      if (res.members.length === 0) {
```
Comment (danielreed) [inline src/listener/server-hardware.js]: Does this need to be changed?

File: src/listener/server-profiles.js [hunk 6]
```diff
@@ -81,7 +81,7 @@ export default class ServerProfilesListener extends Listener {
     let status = msg.status.toLowerCase();
     status = status.charAt(0).toUpperCase() + status.slice(1);
     this.client.ServerProfiles.getProfilesByStatus(status).then((res) => {
-      if (res.count === 0) {
+      if (res.members.length === 0) {
```
Comment (danielreed) [inline src/listener/server-profiles.js]: Does this need to be changed?

File: src/listener/utils/transforms/plain-text-transformer.js [hunk 7]
```diff
@@ -137,6 +140,10 @@ export default class FlowdockTransform {
...
 
   getProviderName() {
-    return 'Flowdock';
+    if (robot.adapterName === 'flowdock') {
+      return 'Flowdock';
+    } else {
+      return 'HipChat';      
+    }
   }
 }
```
Comment (danielreed) [inline src/listener/utils/transforms/plain-text-transformer.js]: Can we name this file just plain-text.js? No need to include transformer in the name of the file, this will match the slack and shell transformers.

File: src/listener/utils/transforms/plain-text-transformer.js [hunk 8]
```diff
@@ -77,7 +77,10 @@ function output(resource) {
   return [];
 }
 
-export default class FlowdockTransform {
+/**
```
Comment (danielreed) [inline src/listener/utils/transforms/plain-text-transformer.js]: I think JS multi-line comments are just /* */

File: src/listener/utils/transforms/plain-text-transformer.js [hunk 9]
```diff
@@ -100,8 +103,8 @@ export default class FlowdockTransform {
 
     const out = output(resource).join('\n');
     if (text) {
-      msg.send(out);
-    }
+      msg.send(text);
```
Comment (danielreed) [inline src/listener/utils/transforms/plain-text-transformer.js]: Aren't we supposed to use out in the msg.send()?

File: src/listener/alerts-listener.js (hunk 1 above)
Comment (llvieira) [reply src/listener/alerts-listener.js]: Basically because the response does not have the count attribute, it has only a member array.

File: src/listener/server-hardware.js (hunk 3 above)
Comment (llvieira) [reply src/listener/server-hardware.js]: Yeah, we do. See the comment above.

File: src/listener/server-hardware.js (hunk 5 above)
Comment (llvieira) [reply src/listener/server-hardware.js]: Yes, see the comment above.

File: src/listener/utils/transforms/plain-text-transformer.js (hunk 9 above)
Comment (llvieira) [reply src/listener/utils/transforms/plain-text-transformer.js]: Yes, we are using yet.

Before:
```javascript

if (text) {
      msg.send(out);
}

msg.send(out);
```

After:
```javascript

if (text) {
      msg.send(text);
}

msg.send(out);
```

The bug was fixed, because before the output was being shown 2 times and the text never.

File: src/listener/ov-listener.js (hunk 2 above)
Comment (llvieira) [reply src/listener/ov-listener.js]: Great, fixed.

File: src/listener/utils/transforms/plain-text-transformer.js (hunk 7 above)
Comment (llvieira) [reply src/listener/utils/transforms/plain-text-transformer.js]: Okay, fixed!

File: src/listener/utils/transforms/plain-text-transformer.js (hunk 8 above)
Comment (llvieira) [reply src/listener/utils/transforms/plain-text-transformer.js]: Fixed!

File: src/listener/server-hardware.js (hunk 4 above)
Comment (llvieira) [reply src/listener/server-hardware.js]: Fixed!
